conans/client/cmd/profile.py

```diff
@@ -21,13 +21,23 @@ def _get_profile_keys(key):
 
 
 def cmd_profile_list(cache_profiles_path, output):
-    folder = cache_profiles_path
-    if os.path.exists(folder):
-        return [name for name in os.listdir(folder)
-                if not os.path.isdir(os.path.join(folder, name))]
-    else:
+    profiles = []
+    if os.path.exists(cache_profiles_path):
+        for current_directory, dirs, files in os.walk(cache_profiles_path):
+            # We don't want to list files from hidden directory
+            dirs[:] = [d for d in dirs if not d.startswith('.')]
+
+            for filename in files:
+                relative_to_profile_dir = os.path.relpath(current_directory, cache_profiles_path)
+                if relative_to_profile_dir == '.':
+                    profiles.append(filename)
+                else:
+                    profiles.append(relative_to_profile_dir + os.sep + filename)
```

Comment:
No need for the if/else

```python
rel_path = os.path.relpath(os.path.join(current_directory, filename), cache_profiles_path)
profiles.append(rel_path)
```

Reply:
@xaqq could you please take a look at this comment?